From 65ba437ed5514dd5762a796d349ed5db49e40fe7 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Wed, 29 Jul 2015 21:26:16 +0200
Subject: [PATCH] platform-config.cmake: fix paths for cross-compilation

Headers and library paths in platform-config.cmake must take into
account ${CMAKE_FIND_ROOT_PATH} to work in cross-compilation.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 platform-config.cmake.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/platform-config.cmake.in b/platform-config.cmake.in
index 7fe35fe..78b5085 100644
--- a/platform-config.cmake.in
+++ b/platform-config.cmake.in
@@ -10,16 +10,16 @@
 #
 # propagate these properties from one build system to the other
 set (platform_VERSION "@platform_VERSION_MAJOR@.@platform_VERSION_MINOR@")
-set (platform_INCLUDE_DIRS @platform_INCLUDE_DIRS@ @CMAKE_INSTALL_PREFIX@/include)
+set (platform_INCLUDE_DIRS ${CMAKE_FIND_ROOT_PATH}/@platform_INCLUDE_DIRS@ ${CMAKE_FIND_ROOT_PATH}/@CMAKE_INSTALL_PREFIX@/include)
 set (platform_LIBRARY_DIRS "@CMAKE_LIBRARY_OUTPUT_DIRECTORY@")
 set (platform_LINKER_FLAGS "@platform_LINKER_FLAGS@")
 set (platform_CONFIG_VARS "@platform_CONFIG_VARS@")
 
 # libraries come from the build tree where this file was generated
 if(WIN32)
-  set (platform_LIBRARY "@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@/platform.lib")
+  set (platform_LIBRARY "${CMAKE_FIND_ROOT_PATH}/@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@/platform.lib")
 else(WIN32)
-  set (platform_LIBRARY "-L@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@ -lplatform")
+  set (platform_LIBRARY "-L${CMAKE_FIND_ROOT_PATH}/@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@ -lplatform")
 endif(WIN32)
 set (platform_LIBRARIES ${platform_LIBRARY} "@platform_LIBRARIES@")
 mark_as_advanced (platform_LIBRARY)
-- 
2.5.0

